lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -501,7 +501,8 @@ public static Domain getDomain(
 
         List<ByteBuffer> maxValues = columnIndex.getMaxValues();
         List<ByteBuffer> minValues = columnIndex.getMinValues();
-        List<Long> nullCounts = columnIndex.getNullCounts();
+        // Null counts is optional in the format, see org.apache.parquet.internal.column.columnindex.ColumnIndexBuilder for reference
+        Optional<List<Long>> nullCounts = Optional.ofNullable(columnIndex.getNullCounts());
```

Comment:
Do we really want to use optionals instead of simple ifs?

Reply:
`ColumnIndex` interface is outside this repo and that doesn't provide `@Nullable` annotation on `columnIndex.getNullCounts`. Using `Optional` ensures that all the code here takes into account the optional nature of this field.